core/src/test/scala/unit/kafka/coordinator/transaction/TransactionCoordinatorTest.scala

```diff
@@ -1185,4 +1213,8 @@ class TransactionCoordinatorTest {
   def errorsCallback(ret: Errors): Unit = {
     error = ret
   }
+
+  def verifyPartitionsInTxnCallback(result: AddPartitionsToTxnResult): Unit = {
+    errors = AddPartitionsToTxnResponse.errorsForTransaction(result.topicResults()).asScala.toMap
```

Comment:
Relying on a global variable is risky here. It would be much better to define the callback within the test itself.